doc/guides/V8-api-for-node.md

```diff
@@ -0,0 +1,210 @@
+    explicitly cleared with Persistent::Reset.
+- Eternal is a persistent handle for JavaScript objects that are expected to
+  never be deleted. It is cheaper to use because it relieves the garbage
+  collector from determining the liveness of that object.
+- A handle scope can be thought of as a container for any number of handles.
+  When you've finished with your handles, instead of deleting each one
+  individually you can simply delete their scope.
+- EscapableHandleScope: Locals can't be returned, their scope will delete them,
+  so you need an escable scope, and to `return scope.Escape(...the local)`. It
+  scopes the locals into the enclosing scope and returns a local for that scope.
+
+Local, MaybeLocal, Maybe, Value, oh my...
+
+https://groups.google.com/forum/#!topic/v8-users/gQVpp1HmbqM
+
+- MaybeLocal may be "empty", basically not contain a pointer of its type.  See
+  `class MaybeLocal`, has some useful notes on why, but basically its returned
+  by anything that might fail to have a value.
+- If you know that the MaybeLocal has a value, then call ToLocalChecked() and
+  node will abort with CHECK() if you are wrong.
+- Otherwise, you have to call `bool ToLocal(Local<S>* out)`, and check the
+  return value to see if there was a value. Or call IsEmpty() to check.
+
+- Maybe is similar, but doesn't hold a Local, just a value of T. "Just" means it
+  "just has a value", a bizarrely named Haskellism :-(. It has a To() and
+  ToChecked() similar to MaybeLocal.
+- A common node idiom is to make a seemingly side-effect free call to
+  `.FromJust()` after `->Set()`, which will crash node if the Set failed.
+  FromJust is also() commonly called  after getting a Maybe<> of a concrete data
+  type from a Local<Value>. It will crash if the conversion fails!
+
+        int32_t v = (Local<Value>)->In32Value(env->context()).FromJust()
+
+- As<T> always returns a value, though probably not one that is useful, its
+  usually best to check the type:
+  - Boolean becomes 1/0 as int, "true"/"false" as strings, etc.
+  - Numbers become false as Boolean (for any value), -3 casts to String "-3"
+  - Functions become numerically zero, and "function () { const hello=0; }" as a
+    String
+  - Examples:
+        CHECK(args[0]->IsInt32());  // Optional, but typical
+        Local<Int32> l = args[0].As<Int32>(); int32_t v = li->Value();
+
+        CHECK(args[0]->IsString());  // Optional, but typical
+        const node::Utf8Value v(env->isolate(), args[0]); const char* s = *s;
+
+- To<T> will convert values in fairly typical js way:
+   ... never seems to be used by node?
+   AFAICT, is identical to the As<> route, except for Boolean, which is always
+   false with As<T>(), but is "expected" with ToT().
+
+- FunctionCallbackInfo
+  - can GetIsolate()
+  - can get Environment: Environment* env = Environment::GetCurrent(args);
+  - can get args using 0-based index
+  - returns Local<Value>
```

Comment:
Maybe fold this into the previous line? It made me think of `args.GetReturnValue().Set(...)` but that isn't what you mean, is it?